src/threading.c

```diff
@@ -960,6 +960,32 @@ JL_DLLEXPORT int jl_alignment(size_t sz)
     return jl_gc_alignment(sz);
 }
 
+JL_DLLEXPORT int jl_getaffinity(int16_t tid, char *mask, int cpumasksize) {
+    int nthreads = jl_atomic_load_acquire(&jl_n_threads);
+    if (tid < 0 || tid >= nthreads)
+        jl_error("invalid tid");
```

Comment:
Maybe `return 0` instead?

And check that ptls2 is not null and if it is null return. I think we can delete foreign threads

Reply:
> Maybe return 0 instead?

We currently return 0 on success (which is what `uv_thread_*affinity` does), so, if anything, we should return `-1` or something else that is `!= 0`. There is also precedence for `jl_error`, that is, I copied it from [here](https://github.com/JuliaLang/julia/blob/9d896dc9f1f105b33aa50ad07bb41d4061e2d439/src/threading.c#L330-L331).

> And check that ptls2 is not null and if it is null return. I think we can delete foreign threads

If we stick with `jl_error` above we should probably also error here. In any case, I'll add a check.

Reply:
I've added errors if `ptls2 == NULL`. Let me know what you think.

Reply:
@vchuravy  Kind bump. I would like to cross this off my to-do list as soon as possible :)

Reply:
I am personally not the biggest fan of mixing errors and return codes. It requires one to both use `try .. catch` and handle the return code.

Reply:
Fair enough. In this case, we would have to define return codes then.

What about this: We only use return codes and define them as follows (i'd put a comment into the source code):

* 0 == success
* 1 == invalid thread id provided
* 2 == `ptls2` was `NULL`
* 3 == `uv_thread_*` failed with a non-zero exit code